### DISCUSSION TITLE
Which state-management solution are you guys using?

### DISCUSSION THREAD

**Discussion by Kay2dan** [blitz-js/blitz#1588] [General]
I started with React's state and now its becoming a serious handicap.

My main `global` state is sitting within `_app.tsx`, that and then the page immediately below are handling most of the state with some states lower down the heirarchy.

I am keen to find what the Blitz community has been using. My natural inclination is to go to [Redux](https://react-redux.js.org) however, I read some good things on [X-state](https://xstate.js.org/) so thinking about that as well.

**Comment by flybayer** [comment]
Imo, here's the modern quadfecta of state management. Each of the four options are very powerful and can all mostly do the same things, but each one is better suited to different cases.

- [xstate](https://xstate.js.org/) — simplifies complex logic (State machines)
- [zustand](https://github.com/pmndrs/zustand) — simplifies flux principles (like Redux)
- [jotai](https://github.com/pmndrs/jotai) — simplifies atomic principles (like Recoil)
- [valtio](https://github.com/pmndrs/valtio) — simplifies proxy principles (like Mobx)

**Reply by Kay2dan** [reply]
Thank you, I didnt know about all these state management solutions & that was exactly the point of this post. Cheers flybayer.
